Objects/unionobject.c

```diff
@@ -284,20 +284,37 @@ is_new_type(PyObject *obj)
     return is_typing_module(obj);
 }
 
+// Emulates short-circuiting behavior of the ``||`` operator
+// while also checking negative values.
+#define CHECK_RES(obj, check) { \
+    int result = check(obj); \
+    if (result < 0) { \
+        goto fail; \
+    } \
+    else if (result > 0) { \
+        return 1; \
+    } \
```

Comment:
It can be simplified:
```suggestion
    if (result) { \
        return result; \
    } \
```

Reply:
Oh yeah. Thanks. I feel a little silly for not seeing that 😆 .